src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -129,6 +136,95 @@ public void workspaceAuthz(AuthenticatedUserRequest userReq, UUID workspaceId, S
               userReq.getEmail(), action, workspaceId.toString()));
   }
 
+  /**
+   * Wrapper around Sam client to grant a role to the provided user.
+   *
+   * <p>This operation is only available to MC_TERRA stage workspaces, as Rawls manages permissions
+   * directly on other workspaces.
+   *
+   * @param workspaceId The workspace this operation takes place in
+   * @param userReq Credentials of the user requesting this operation. Only owners have permission
+   *     to modify roles in a workspace.
+   * @param role The role being granted.
+   * @param email The user being granted a role.
+   */
+  public void grantWorkspaceRole(
+      UUID workspaceId, AuthenticatedUserRequest userReq, IamRole role, String email) {
+    WorkspaceStage stage = workspaceDao.getWorkspaceStage(workspaceId);
+    if (!WorkspaceStage.MC_WORKSPACE.equals(stage)) {
+      throw new StageDisabledException(workspaceId, stage, "addWorkspaceRole");
+    }
+
+    ResourcesApi resourceApi = samResourcesApi(userReq.getRequiredToken());
+    try {
+      resourceApi.addUserToPolicy(
+          SamUtils.SAM_WORKSPACE_RESOURCE, workspaceId.toString(), role.toSamRole(), email);
+      logger.info(
+          String.format(
+              "Granted role %s to user %s in workspace %s",
+              role.toSamRole(), email, workspaceId.toString()));
+    } catch (ApiException e) {
+      throw new SamApiException(e);
+    }
+  }
+
+  /**
+   * Wrapper around Sam client to remove a role from the provided user.
+   *
+   * <p>This operation is only available to MC_TERRA stage workspaces, as Rawls manages permissions
+   * directly on other workspaces.
+   * Trying to remove a role that a user does not have will succeed, though Sam will error if the
+   * email is not a registered user.
+   */
+  public void removeWorkspaceRole(
+      UUID workspaceId, AuthenticatedUserRequest userReq, IamRole role, String email) {
+    WorkspaceStage stage = workspaceDao.getWorkspaceStage(workspaceId);
+    if (!WorkspaceStage.MC_WORKSPACE.equals(stage)) {
```

Comment:
Just so I understand... since it is an enum, can't you use `==` ?

Reply:
yep, either works. I usually default to `equals`, is there a reason to use one over the other?